nixos/modules/config/update-users-groups.pl

```diff
@@ -281,3 +281,32 @@ sub parseUser {
 }
 
 updateFile("/etc/shadow", \@shadowNew, 0600);
+
+# Rewrite /etc/subuid & /etc/subgid to include default container mappings
+my @subGids;
+my @subUids;
+foreach my $u (values %usersOut) {
+    my $name = $u->{name};
+
+    foreach my $range (@{$u->{subUidRanges}}) {
+        my $value = join(":", ($name, $range->{startUid}, $range->{count}));
+        push @subUids, $value;
+    }
+
+    foreach my $range (@{$u->{subGidRanges}}) {
+        my $value = join(":", ($name, $range->{startGid}, $range->{count}));
+        push @subGids, $value;
+    }
+
+    if($u->{isNormalUser}) {
+        my $offset = ($u->{uid} - 1000 + 1);
```

Comment:
What happens if someone creates a normal user with a UID of 666?

Reply:
I've fixed this by not infering the subuid range from a UID but instead using the same approach as UID/GID allocation is using.